### ISSUE TITLE
GetMessage should return int

### ISSUE THREAD

**Issue by AArnott** [microsoft/win32metadata#308]
> There are APIs that return more than two values (most prominently [GetMessage](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getmessage#return-value)), in case thats relevant to the decision

_Originally posted by @weltkante in https://github.com/microsoft/win32metadata/issues/301#issuecomment-791279429_

From `GetMessage` docs:

> Because the return value can be nonzero, zero, or -1, avoid code like this:

This is a bit hard to follow, since it apparently doesn't classify `-1` as "nonzero", which in fact it is.
How about we break away from the metadata and force this method to return `int` since it doesn't limit its return value to just zero and non-zero? In fact the docs themselves warn against code like this: `while (GetMessage( lpMsg, hWnd, 0, 0))`, so forcing the return type to be int would help guide folks to follow the docs more.

**Comment by sotteson1** [comment]
I'm not sure. A BOOL doesn't restrict itself to 1 or 0. It's an int, just like the Win32 BOOL.

**Comment by kennykerr** [comment]
Yes, there are a few functions that do this and `BOOL` already leaves room for it. 

Incidentally, I generally just ignore the result of `GetMessage`. It's simpler to check the resulting message:

https://github.com/microsoft/windows-rs/blob/8d84c192afd45b6094d53eb23e6962b2b5b782b7/examples/clock/src/main.rs#L456-L460

**Comment by weltkante** [comment]
> Incidentally, I generally just ignore the result of GetMessage.

If you only pass zero/null arguments then it [can't fail](https://devblogs.microsoft.com/oldnewthing/20130322-00/?p=4873) anyways

**Comment by sotteson1** [comment]
I'm going to close this because I think we should preserve the API signature the way it's defined. The way BOOL works in Win32 isn't obvious to those used to bool which is either 1 or 0, but it's how Win32 has always worked.

**Comment by AArnott** [comment]
Ok. In CsWin32 I can make sure `BOOL` provides access to the underlying `int` while still allowing implicit conversion to `bool` based on a non-zero value.
